rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -587,6 +587,26 @@ __find_parameter_by_name(
     [&](auto parameter) {return parameter.get_name() == name;});
 }
 
+rcl_interfaces::msg::SetParametersResult
+NodeParameters::set_parameters_atomically_helper(const std::vector<rclcpp::Parameter> & parameters){
+  const auto& result = this->set_parameters_atomically(parameters);
+
+  if(!on_local_parameters_set_callback_container_.empty() && result.successful){
+    auto it = on_local_parameters_set_callback_container_.begin();
+    while (it != on_local_parameters_set_callback_container_.end()) {
+      auto shared_handle = it->lock();
+      if (nullptr != shared_handle) {
+        shared_handle->callback(parameters);
+        it++;
+      } else {
+        it = on_local_parameters_set_callback_container_.erase(it);
+      }
+    }
+  }
+
+  return result;
```

Comment:
One more concern here that might be worth discussing : When we set a parameter in the command line say using
``ros2 param set /some_node some_parameter False``, I want it to return after the validation call ends. According to the current implementation, it will block until the new "pose-set" callbacks have returned, @deepanshubansal01 can correct me if I'm wrong.

Ideally I would want it to return after the validation call is done. @jacobperron what do you think of am async implementation ?